logic.py

```diff
@@ -81,9 +82,10 @@ def tell(self, sentence):
         self.clauses.extend(conjuncts(to_cnf(sentence)))
 
     def ask_generator(self, query):
-        "Yield the empty substitution if KB implies query; else nothing."
+        "Return the empty substitution {} if KB entails query; else return False."
         if tt_entails(Expr('&', *self.clauses), query):
-            yield {}
+            yield {}  # Why use yield when you are not returning a generator?
```

Comment:
Yes, that sounds like a good idea. So, we can encourage a first-time reader who is just getting started with Propositional Logic to use `ask_if_true` instead of `ask` or `ask_generator` so that she doesn't get confused. I'll make the necessary changes to the PR.